glslc/src/dependency_info.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015 The Shaderc Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GLSLC_DEPENDENCY_INFO_H
+#define GLSLC_DEPENDENCY_INFO_H
+
+#include <set>
+#include <string>
+
+namespace glslc {
+class DependencyInfoDumpingHandler {
+ public:
+  // Sets -MT argument.
+  void SetDepTargetFileLabel(const std::string& target_label) {
+    dep_target_label_ = std::string(target_label);
+  }
+
+  // Sets -MF argument.
+  void SetDepFileName(const std::string& dep_file_name) {
+    dep_file_name_ = dep_file_name;
+  };
+
+  // Collects the dependency info and dump it to dependency info file.
```

Comment:
Does it do the collection, or does it just write the info.
Why do you need to set the target file if there's a separate method to do it?
What's the trace argument for?  Why is it called "trace"?